library/src/main/java/com/bumptech/glide/request/SingleRequest.java

```diff
@@ -387,7 +387,8 @@ public boolean isAnyResourceSet() {
   private Drawable getErrorDrawable() {
     if (errorDrawable == null) {
       errorDrawable = requestOptions.getErrorPlaceholder();
-      if (errorDrawable == null && requestOptions.getErrorId() > 0) {
+      if (errorDrawable == null && requestOptions.getErrorId() != -1 &&
+              (requestOptions.getErrorId() & 0xff000000) != 0 && (requestOptions.getErrorId() & 0x00ff0000) != 0) {
```

Comment:
```java
/**
 * @see android.content.res.ResourceId.isValid
 */
private static boolean isValidId(int id) {
  return id != -1 && id & 0xff000000 != 0 && id  & 0x00ff0000 != 0;
}

if (isValidId(requestOptions.getErrorId()))
if (isValidId(requestOptions.getPlaceholderId()))
if (isValidId(requestOptions.getFallbackId()))
```
Can we reduce duplicated code like above?

Reply:
Thanks! I have made the changes accordingly. 😊